aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecorator.kt

```diff
@@ -0,0 +1,172 @@
+import software.amazon.smithy.rust.codegen.client.smithy.generators.OperationCustomization
+import software.amazon.smithy.rust.codegen.client.smithy.generators.OperationSection
+import software.amazon.smithy.rust.codegen.client.smithy.generators.client.FluentClientGenerator
+import software.amazon.smithy.rust.codegen.core.rustlang.Attribute
+import software.amazon.smithy.rust.codegen.core.rustlang.Attribute.Companion.cfg
+import software.amazon.smithy.rust.codegen.core.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.core.rustlang.Writable
+import software.amazon.smithy.rust.codegen.core.rustlang.rust
+import software.amazon.smithy.rust.codegen.core.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.core.rustlang.writable
+import software.amazon.smithy.rust.codegen.core.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.core.smithy.generators.BuilderGenerator
+import software.amazon.smithy.rust.codegen.core.smithy.generators.Instantiator
+import software.amazon.smithy.rust.codegen.core.smithy.generators.setterName
+import software.amazon.smithy.rust.codegen.core.util.dq
+import software.amazon.smithy.rust.codegen.core.util.expectTrait
+import software.amazon.smithy.rust.codegen.core.util.inputShape
+import software.amazon.smithy.rust.codegen.core.util.letIf
+import software.amazon.smithy.rust.codegen.core.util.orNull
+import software.amazon.smithy.rust.codegen.core.util.toSnakeCase
+import software.amazon.smithy.smoketests.traits.Expectation
+import software.amazon.smithy.smoketests.traits.SmokeTestCase
+import software.amazon.smithy.smoketests.traits.SmokeTestsTrait
+import java.util.Optional
+
+class SmokeTestsDecorator : ClientCodegenDecorator {
+    override val name: String = "SmokeTests"
+    override val order: Byte = 0
+
+    override fun operationCustomizations(
+        codegenContext: ClientCodegenContext,
+        operation: OperationShape,
+        baseCustomizations: List<OperationCustomization>,
+    ): List<OperationCustomization> =
+        baseCustomizations.letIf(operation.hasTrait(SmokeTestsTrait.ID)) {
+            it + SmokeTestsOperationCustomization(codegenContext, operation)
+        }
+}
+
+class SmokeTestsOperationCustomization(
+    private val codegenContext: ClientCodegenContext,
+    private val operation: OperationShape,
+) : OperationCustomization() {
+    private val smokeTestsTrait = operation.expectTrait<SmokeTestsTrait>()
+    private val testCases: List<SmokeTestCase> = smokeTestsTrait.testCases
+    private val operationInput = operation.inputShape(codegenContext.model)
+
+    override fun section(section: OperationSection): Writable {
+        if (testCases.isEmpty()) return emptySection
+
+        return when (section) {
+            is OperationSection.UnitTests ->
+                writable {
+                    testCases.forEach { testCase ->
+                        Attribute(cfg("v2SmokeTests")).render(this)
```

Comment:
**question**: Why `v2SmokeTests`? Why not just `smoketests`?

Reply:
We have multiple conceptions of what a smoke test is in the SDK and I decided to be specific to make things less ambiguous.

Previously, we've referred to the models included in smithy-rs as "smoke test models" and I wanted to associate this feature with the related internal designation instead.

Reply:
I think my issue is that these are associated with the [smokeTests trait](https://smithy.io/2.0/additional-specs/smoke-tests.html#smithy-test-smoketests-trait) which has no connection to a "v2SmokeTests". 

I also don't see how "v2SmokeTests" improves the status quo since it implies there was a v1 at some point. I think we could solve this pretty easily with some docs on the generated module if you take the suggestion to group them all into a common module.

Reply:
I changed it to `smoketests`